sys/ps/ps.c

```diff
@@ -43,6 +43,28 @@ static const char *state_names[] = {
     [STATUS_COND_BLOCKED] = "bl cond",
 };
 
+#define STATE_NAMES_NUMOF ((int)(sizeof(state_names)/sizeof(state_names[0])))
```

Comment:
`STATUS_NUMOF` gixves the number of possible thread states. It would be possible to just set the number of array elements in `state_names` to `STATUS_NUMOF` so the compiler would initialize all not explicitly set elements with `NULL`. Then it can be (almost) safely assumed that `state_names[x]` will be either the correct state name or `NULL`, assuming that `x` is a valid enum value (and not `STATUS_NUMOF`). This assumption could simplify the `state_to_string()` function a bit

Reply:
Maybe it is better to keep `STATE_NAMES_NUMOF` and have something like:

```C
_Static_assert(STATE_NAMES_NUMOF == STATUS_NUMOF, "Missing entry in state_names")
```

This would be a compile time check like @kaspar030 suggested and for any non-correct `thread_t` data structure it would then be sufficient to use the current stringification code. (That is, not adding a wrapper function to save ROM.)

Reply:
@maribu I agree with the first comment. The second suggestion would be good, too, but _Static_assert is C11 and afaik we try to keep riot in c99 (even if we use some GCC extension like in the initialization of state_names)